plugins/inputs/solr/solr.go

```diff
@@ -0,0 +1,424 @@
+		AvgRequestsPerSecond     float64 `json:"avgRequestsPerSecond"`
+		AvgTimePerRequest        float64 `json:"avgTimePerRequest"`
+		Errors                   int     `json:"errors"`
+		HandlerStart             int     `json:"handlerStart"`
+		MedianRequestTime        float64 `json:"medianRequestTime"`
+		Requests                 int     `json:"requests"`
+		Timeouts                 int     `json:"timeouts"`
+		TotalTime                float64 `json:"totalTime"`
+	} `json:"stats"`
+}
+
+// UpdateHandler is an exported type that
+// contains update handler metrics
+type UpdateHandler struct {
+	Class string `json:"class"`
+	Stats struct {
+		Adds                     int    `json:"adds"`
+		AutocommitMaxDocs        int    `json:"autocommit maxDocs"`
+		AutocommitMaxTime        string `json:"autocommit maxTime"`
+		Autocommits              int    `json:"autocommits"`
+		Commits                  int    `json:"commits"`
+		CumulativeAdds           int    `json:"cumulative_adds"`
+		CumulativeDeletesByID    int    `json:"cumulative_deletesById"`
+		CumulativeDeletesByQuery int    `json:"cumulative_deletesByQuery"`
+		CumulativeErrors         int    `json:"cumulative_errors"`
+		DeletesByID              int    `json:"deletesById"`
+		DeletesByQuery           int    `json:"deletesByQuery"`
+		DocsPending              int    `json:"docsPending"`
+		Errors                   int    `json:"errors"`
+		ExpungeDeletes           int    `json:"expungeDeletes"`
+		Optimizes                int    `json:"optimizes"`
+		Rollbacks                int    `json:"rollbacks"`
+		SoftAutocommits          int    `json:"soft autocommits"`
+	} `json:"stats"`
+}
+
+// Cache is an exported type that
+// contains cache metrics
+type Cache struct {
+	Class string `json:"class"`
+	Stats struct {
+		CumulativeEvictions int     `json:"cumulative_evictions"`
+		CumulativeHitratio  float64 `json:"cumulative_hitratio,string"`
+		CumulativeHits      int     `json:"cumulative_hits"`
+		CumulativeInserts   int     `json:"cumulative_inserts"`
+		CumulativeLookups   int     `json:"cumulative_lookups"`
+		Evictions           int     `json:"evictions"`
+		Hitratio            float64 `json:"hitratio,string"`
+		Hits                int     `json:"hits"`
+		Inserts             int     `json:"inserts"`
+		Lookups             int     `json:"lookups"`
+		Size                int     `json:"size"`
+		WarmupTime          int     `json:"warmupTime"`
+	} `json:"stats"`
+}
+
+// NewSolr return a new instance of Solr
+func NewSolr() *Solr {
+	return &Solr{
+		HTTPTimeout: internal.Duration{Duration: time.Second * 5},
+	}
+}
+
+// SampleConfig returns sample configuration for this plugin.
+func (s *Solr) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description returns the plugin description.
+func (s *Solr) Description() string {
+	return "Read stats from one or more Solr servers or cores"
+}
+
+// Default settings
+func (s *Solr) setDefaults() ([][]string, int, error) {
```

Comment:
Perhaps a better name for this function? It doesn't seem to be dealing much with defaults, just getting a list of cores to enumerate.